pkg/healthcheck/healthcheck.go

```diff
@@ -1270,17 +1412,37 @@ func (hc *HealthChecker) checkClusterRoles(shouldExist bool) error {
 
 	for _, item := range crList.Items {
 
-		if hc.skipNoInitContainerResources(item.ObjectMeta.Labels) {
+		if skipCni && isCniResource(item.ObjectMeta.Labels) {
```

Comment:
So this piece of logic was introduced some time ago because of the situation described right [here](https://github.com/linkerd/linkerd2/pull/3066#issuecomment-512479059). 

So originally this logic was just skipping resources that were CNI when CNI was enabled to check resources that are not CNI. Now however, we want to use this logic to check resources  that are not CNI when CNI is enabled and disabled, we also want to use it to check CNI resources when CNI is enabled. Therefore the ectra boolean that could be passed in depending on the check we are performing. Reading it, I see it does have some cognitive luggage indeed. Aghh... how soething so simple turned out to be so unnecessarily complicated....